### PR TITLE
Add FuseMLP to NPU Llama

### PR DESCRIPTION
**Pull request by hkvision**

### CODE REVIEW

File: python/llm/src/ipex_llm/transformers/npu_models/fusedmlp.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
+from intel_npu_acceleration_library.backend.factory import NNFactory
```
Comment (jason-dai) [inline python/llm/src/ipex_llm/transformers/npu_models/fusedmlp.py:1]: change to `lowbitmlp.py`

File: python/llm/src/ipex_llm/transformers/npu_models/fusedmlp.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+from intel_npu_acceleration_library.backend.runtime import set_contiguous, record_function, adapt_output_tensor, _model_cache
+from typing import Optional, Sequence, List, Union, Any
+from functools import partial
+from collections import deque
+import numpy as np
+import torch
+import uuid
+
+
+class QuantizedMLP(NNFactory):
```
Comment (jason-dai) [inline python/llm/src/ipex_llm/transformers/npu_models/fusedmlp.py:11]: change to `LowBitMLP`

File: python/llm/src/ipex_llm/transformers/npu_models/fusedmlp.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+        original_shape = x.shape
+        if len(x.shape) > 2:
+            x = x.view([-1, x.shape[-1]])
+        output = run_factory(x, self.op_parameters, self.backend_cls, self.op_id)
+        return output.view(original_shape)
+
+
+# TODO: separate it into a single file
+@torch.no_grad()
+def run_factory(
```
Comment (jason-dai) [inline python/llm/src/ipex_llm/transformers/npu_models/fusedmlp.py:104]: change to `run_model`

File: python/llm/src/ipex_llm/transformers/npu_models/fusedmlp.py [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+    x: Union[torch.Tensor, List[torch.Tensor]],
+    weights: List[torch.Tensor],
+    backend_cls: Any,
+    op_id: Optional[str] = None,
+) -> torch.Tensor:
+    """Run a factory operation. Depending on the datatype of the weights it runs a float or quantized operation.
```
Comment (jason-dai) [inline python/llm/src/ipex_llm/transformers/npu_models/fusedmlp.py:110]: Run an NPU model

File: python/llm/src/ipex_llm/transformers/npu_models/fusedmlp.py [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+
+    shape_dtype_signature = "_".join(
+        ["_".join(str(dim) for dim in t.shape) + f"_{t.dtype}" for t in x_np + op_args_flatten]
+    )
+    key = f"{backend_cls.func.__name__}_{shape_dtype_signature}"
+    models = _model_cache.get(key, None)
+
+    input_shapes = [elem.shape for elem in x_np]
+    if models is None:
+        _model_cache[key] = deque([backend_cls(*input_shapes) for i in range(4)])
```
Comment (jason-dai) [inline python/llm/src/ipex_llm/transformers/npu_models/fusedmlp.py:151]: make 4 configurable

File: python/llm/src/ipex_llm/transformers/npu_models/fusedmlp.py (hunk 1 above)
Comment (hkvision) [reply python/llm/src/ipex_llm/transformers/npu_models/fusedmlp.py:1]: sure.

**Pull request closed without merging**